utoipa-gen/src/schema/component.rs

```diff
@@ -690,47 +791,57 @@ where
 }
 
 #[inline]
-fn is_not_skipped(rule: &Option<Serde>) -> bool {
+fn is_not_skipped(rule: &Option<SerdeValue>) -> bool {
     rule.as_ref()
-        .map(|rule| matches!(rule, Serde::Value(value) if value.skip == None))
+        .map(|value| value.skip.is_none())
         .unwrap_or(true)
 }
 
+/// Resolves the appropriate [`RenameRule`] to apply to the specified `struct` `field` name given a
+/// `container_rule` (`struct` or `enum` level) and `field_rule` (`struct` field or `enum` variant
+/// level). Returns `Some` of the result of the `rename_op` if a rename is required by the supplied
+/// rules.
 #[inline]
 fn rename_field<'a>(
-    container_rule: &'a mut Option<Serde>,
-    field_rule: &'a mut Option<Serde>,
+    container_rule: &'a mut Option<SerdeContainer>,
+    field_rule: &'a mut Option<SerdeValue>,
     field: &str,
 ) -> Option<String> {
     rename(container_rule, field_rule, &|rule| rule.rename(field))
 }
 
+/// Resolves the appropriate [`RenameRule`] to apply to the specified `enum` `variant` name given a
+/// `container_rule` (`struct` or `enum` level) and `field_rule` (`struct` field or `enum` variant
+/// level). Returns `Some` of the result of the `rename_op` if a rename is required by the supplied
+/// rules.
 #[inline]
 fn rename_variant<'a>(
-    container_rule: &'a mut Option<Serde>,
-    field_rule: &'a mut Option<Serde>,
-    field: &str,
+    container_rule: &'a mut Option<SerdeContainer>,
+    field_rule: &'a mut Option<SerdeValue>,
+    variant: &str,
 ) -> Option<String> {
     rename(container_rule, field_rule, &|rule| {
-        rule.rename_variant(field)
+        rule.rename_variant(variant)
     })
 }
 
+/// Resolves the appropriate [`RenameRule`] to apply during a `rename_op` given a `container_rule`
+/// (`struct` or `enum` level) and `field_rule` (`struct` field or `enum` variant level). Returns
+/// `Some` of the result of the `rename_op` if a rename is required by the supplied rules.
 #[inline]
 fn rename<'a>(
-    container_rule: &'a mut Option<Serde>,
-    field_rule: &'a mut Option<Serde>,
+    container_rule: &'a mut Option<SerdeContainer>,
+    field_rule: &'a mut Option<SerdeValue>,
     rename_op: &impl Fn(&RenameRule) -> String,
 ) -> Option<String> {
-    let rename = |rule: &mut Serde| match rule {
-        Serde::Container(container) => container.rename_all.as_ref().map(rename_op),
-        Serde::Value(ref mut value) => mem::take(&mut value.rename),
-    };
-
     field_rule
         .as_mut()
-        .and_then(rename)
-        .or_else(|| container_rule.as_mut().and_then(rename))
+        .and_then(|value| value.rename.take())
+        .or_else(|| {
+            container_rule
+                .as_mut()
+                .and_then(|container| container.rename_all.as_ref().map(rename_op))
+        })
```

Comment:
:+1:  It's a clearer now, The `&mut`' could be now removed as well to make it even more clear since they existed there only for the `men::take` which is not used any more. :)
```rust
Serde::Value(ref mut value) => mem::take(&mut value.rename),
```